docs/topics/tour/kotlin-tour-intermediate-null-safety.md

```diff
@@ -0,0 +1,658 @@
+[//]: # (title: Intermediate: Null safety)
+
+In the beginner's tour, you learned how to handle `null` values in your code but there are also use cases where using 
+null safety features can be helpful.
+
+## Smart casts and safe casts
+
+You already know how to explicitly declare the type of a variable or object and that Kotlin sometimes has the ability
+to infer the type without explicit declaration. When you tell Kotlin to treat a variable or object as if it belongs to a
+specific type, this process is called **casting**. When a type is automatically cast, like when it's inferred, it's called
+**smart casting**.
+
+Before we explore how casting works, how can you check if an object has a certain type? For this, you can use the
```

Comment:
Just a nitpick, I feel like the connection in the first sentence is not very clear, what about:

> Before we explore how casting works, let's learn how you can check if an object has a certain type.

or

> Before exploring how casting works, let's see how to check if an object has a certain type.